livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/stt.py

```diff
@@ -127,44 +137,69 @@ def __init__(
+                    samples_per_second=self._opts.sample_rate,
+                    bits_per_sample=16,
+                    channels=self._opts.num_channels,
+                )
             )
-        )
-        self._recognizer = _create_speech_recognizer(
-            config=self._opts, stream=self._stream
-        )
-        self._recognizer.recognizing.connect(self._on_recognizing)
-        self._recognizer.recognized.connect(self._on_recognized)
-        self._recognizer.speech_start_detected.connect(self._on_speech_start)
-        self._recognizer.speech_end_detected.connect(self._on_speech_end)
-        self._recognizer.session_started.connect(self._on_session_started)
-        self._recognizer.session_stopped.connect(self._on_session_stopped)
-        self._recognizer.start_continuous_recognition()
-
-        try:
-            await asyncio.wait_for(
-                self._session_started_event.wait(), self._conn_options.timeout
+            self._recognizer = _create_speech_recognizer(
+                config=self._opts, stream=self._stream
             )
-
-            async for input in self._input_ch:
-                if isinstance(input, rtc.AudioFrame):
-                    self._stream.write(input.data.tobytes())
-
-            self._stream.close()
-            await self._session_stopped_event.wait()
-        finally:
-
-            def _cleanup():
-                self._recognizer.stop_continuous_recognition()
-                del self._recognizer
-
-            await asyncio.to_thread(_cleanup)
+            self._recognizer.recognizing.connect(self._on_recognizing)
+            self._recognizer.recognized.connect(self._on_recognized)
+            self._recognizer.speech_start_detected.connect(self._on_speech_start)
+            self._recognizer.speech_end_detected.connect(self._on_speech_end)
+            self._recognizer.session_started.connect(self._on_session_started)
+            self._recognizer.session_stopped.connect(self._on_session_stopped)
+            self._recognizer.start_continuous_recognition()
+
+            try:
+                await asyncio.wait_for(
+                    self._session_started_event.wait(), self._conn_options.timeout
+                )
+
+                async def process_input():
+                    async for input in self._input_ch:
+                        if isinstance(input, rtc.AudioFrame):
+                            self._stream.write(input.data.tobytes())
+
+                async def _wait_for_reconnect():
+                    await self._reconnect_event.wait()
+
+                process_input_task = asyncio.create_task(process_input())
+                reconnect_task = asyncio.create_task(_wait_for_reconnect())
+                try:
+                    await asyncio.wait(
+                        [process_input_task, reconnect_task],
+                        return_when=asyncio.FIRST_COMPLETED,
+                    )
+                except asyncio.CancelledError:
```

Comment:
I think it's risky to catch the CancelledError here, it may come from the baseclass